src/S3/MultipartUploadingTrait.php

```diff
@@ -54,10 +54,17 @@ protected function handleResult(CommandInterface $command, ResultInterface $resu
         $partData = [];
         $partData['PartNumber'] = $command['PartNumber'];
         $partData['ETag'] = $this->extractETag($result);
+
+        $checksumResult = $this instanceof MultipartCopy
+            ? $result['CopyPartResult']
+            : $result;
         if (isset($command['ChecksumAlgorithm'])) {
             $checksumMemberName = 'Checksum' . strtoupper($command['ChecksumAlgorithm']);
-            $partData[$checksumMemberName] = $result[$checksumMemberName];
+            $partData[$checksumMemberName] = $checksumResult[$checksumMemberName];
```

Comment:
Is it guaranteed that `$checksumResult` will contain a member `$checksumMemberName`?

Reply:
Yes- if a command checksum is provided, we can assume it was valid if we're handling the result.  S3 will send the corresponding encoded value in the `$checksumMemberName` result key.